Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py

```diff
@@ -220,19 +228,112 @@ def search_records_by_report_soap_request(token, report_guid):
+    field_name: str,
+    field_id: str,
+    search_value: str,
+    sub_elements_tags_values: dict[str, Any] | None = None
+) -> str:
+    """
+    Constructs an XML string representing any generic filter condition for searching records.
+
+    Args:
+        condition_type (FilterConditionTypes): The type of the filter condition.
+        operator (str): The comparison operator (e.g., 'Equals', 'GreaterThan').
+        field_name (str): The name of the application field.
+        field_id (str): The ID of the application field.
+        search_value (str): The value for the comparison.
+        sub_elements_tags_values (dict | None): Optional tag names and values to embed as sub-elements.
+
+    Returns:
+        str: An XML string representing the FilterCondition element.
+    """
+    root = ET.Element(condition_type.value)
+
+    operator_element = ET.SubElement(root, 'Operator')
+    operator_element.text = operator
+
+    field_element = ET.SubElement(root, 'Field', attrib={'name': field_name})
+    field_element.text = field_id
+
+    value_element = ET.SubElement(root, 'Value')
+    value_element.text = search_value
+
+    for tag_name, tag_text in (sub_elements_tags_values or {}).items():
+        sub_element = ET.SubElement(root, tag_name)
+        sub_element.text = tag_text
+
+    return ET.tostring(root, encoding='unicode')
+
+
+def construct_content_filter_condition(operator: str, level_id: str, search_value: str) -> str:
+    """
+    Constructs an XML string representing a content filter condition for searching records.
+
+    Args:
+        operator (str): The comparison operator (e.g., 'Equals', 'GreaterThan').
+        field_name (str): The name of the application field.
+        field_id (str): The ID of the application field.
+        search_value (str): The value for the comparison.
+
+    Returns:
+        str: An XML string representing the ContentFilterCondition element.
+    """
+    root = ET.Element(FilterConditionTypes.content.value)
+
+    level = ET.SubElement(root, 'Level')
+    level.text = level_id
+
+    operator_element = ET.SubElement(root, 'Operator')
+    operator_element.text = operator
+
+    values_element = ET.SubElement(root, 'Values')
+    value_element = ET.SubElement(values_element, 'Value')
+    value_element.text = search_value
+
+    return ET.tostring(root, encoding='unicode')
+
+
+def construct_operator_logic(logical_operator: str | None, conditions_count: int) -> str:
+    """
+    Constructs an XML string representing the operator logic that applies to the filtering conditions for searching records.
+
+    Args:
+        logical_operator (str | None): The logical operator (e.g. 'AND', 'OR', 'XOR').
+        conditions_count (int): The number of conditions.
+
+    Returns:
+        str: An XML string representing the OperatorLogic element.
+    """
+    if not logical_operator or not logical_operator.strip():
+        return ''
+
+    # Only relevant if there are two or more conditions
```

Comment:
why?

Reply:
The reason for it is actually quite ... _**logical**_ (no pun intended 😉)

I have improved the comment in commit 1e06e3deab3a1b2370083538269f7a6ae491e32c

It now says:
> _Only two or more conditions can be joined by a logical operator (e.g. '1 AND 2', '1 OR 2 OR 3')_

Let me know if you want me to further explain this.

Reply:
should less than 2 be an error?

Reply:
We could raise an exception here (e.g. `ValueError`), but we would need to avert it or catch it in all places where this function is called and set an empty string.

In my opinion, handling everything inside this function yields cleaner code.

Let me know which method you prefer.

Reply:
If we just raised it with a good value, it'll propagate and show (as in all integrations)

If we return `''`, what will happen?

Reply:
If we make a request with the `OperatorLogic` tag and we have fewer than two conditions, the server responds with a HTTP 400 status code [bad request], so the empty string `''` is actually  the desired outcome in this case.

To improve the visibility / monitoring of the logic of this function, we could:
* **Option A**: Raise an exception inside the function and catch it where needed.
* **Option B**: Log a `demisto.debug` or `demisto.info` message under the `if` statement in this function.

Let me know, which option you prefer.

Reply:
let's raise here, make sure it's caught in the `main`